decompiler/backend/cexpressiongenerator.py

```diff
@@ -15,11 +15,46 @@
 )
 from decompiler.structures.pseudo import instructions as instructions
 from decompiler.structures.pseudo import operations as operations
+from decompiler.structures.pseudo.complextypes import Struct
 from decompiler.structures.pseudo.operations import MemberAccess
 from decompiler.structures.visitors.interfaces import DataflowObjectVisitorInterface
 from decompiler.util.integer_util import normalize_int
 
 MAX_GLOBAL_INIT_LENGTH = 128
+INLINE_STRUCT_STRINGS = True
+DETECT_STRUCT_STRINGS = True
```

Comment:
Yes. Because `CExpressionGenerator` does not allow any options, I put them there. The behavior they control affects only `CExpressionGenerator` and not `CodeVisitor`.
What solution would you suggest?

Reply:
Ah I see than you are at the wrong place.
`CExpressionGenerator ` is supposed to be the fall-back, barebones version not reuireing a task object.
If you need options, you want to implement this code in `decompiler/backend
/codevisitor.py` which actually overrides the `CExpressionGenerator `.

Reply:
Moving this code would probably require redesigning parts of the backend. My suggestion would be:
 We could pass an instance of an `CExpressionGenerator` (which could also be a configured `CodeVisitor`) to  `GlobalDeclarationGenerator`) when it is initialized. The functions for determining whether s.th. should be inlined could then become methods of `CExpressionGenerator` which are overwritten in `CodeVisitor`.

Should I follow through with this?

---

The background why this is tricky...:
The options concern inlining, which is a behavior that affects both the `GlobalDeclarationGenerator` and `CExpressionGenerator`. Having a default in `CExpressionGenerator` and options in `CodeVisitor` comes with problems:
- `GlobalDeclarationGenerator` (which does not use a task object) will not know about the options
- passing options to it would require somehow merging options of multiple tasks
- the functions that determine inlining need to become methods of `CodeVisitor` to access options
- We could use `CExpressionGenerator` within `GlobalDeclarationGenerator` anymore, as it is not properly configured.